cmd/bootkube/start.go

```diff
@@ -22,21 +22,31 @@ var (
 		assetDir        string
 		podManifestPath string
 		strict          bool
+		requiredPods    []string
 	}
 )
 
+var defaultRequiredPods = []string{
+	"pod-checkpointer",
+	"kube-apiserver",
+	"kube-scheduler",
+	"kube-controller-manager",
+}
+
 func init() {
 	cmdRoot.AddCommand(cmdStart)
 	cmdStart.Flags().StringVar(&startOpts.assetDir, "asset-dir", "", "Path to the cluster asset directory. Expected layout genereted by the `bootkube render` command.")
 	cmdStart.Flags().StringVar(&startOpts.podManifestPath, "pod-manifest-path", "/etc/kubernetes/manifests", "The location where the kubelet is configured to look for static pod manifests.")
 	cmdStart.Flags().BoolVar(&startOpts.strict, "strict", false, "Strict mode will cause bootkube to exit early if any manifests in the asset directory cannot be created.")
+	cmdStart.Flags().StringSliceVar(&startOpts.requiredPods, "required-pods", defaultRequiredPods, "List of pods (unqualified for pods in kube-system namespace or <ns>/<pod> for arbitrary namespaces) that are required to be running before the start command does the pivot.")
```

Comment:
nit: I'd probably lean toward always being explicit with namespace. And even more minor nit: s/ns/namespace/